projects/igniteui-angular/src/lib/drop-down/drop-down.component.ts

```diff
@@ -524,6 +524,13 @@ export class IgxDropDownComponent extends IgxDropDownBaseDirective implements ID
         }
     }
 
+    /**
+     * @hidden @internal
+     */
+    public listClass() {
```

Comment:
Getter?

Reply:
I think I would either change the return signature of this method to be an object with the keys holding the class names where the values are expressions that evaluate to boolean based on whether the list has items in it or not.
The other option is to make a getter like @kdinev suggested, which returns the length of the list and to which the `--empty` class gets host-bound.

Reply:
@kdinev, @simeonoff I've now applied the comments from @rkaraivanov: https://github.com/IgniteUI/igniteui-angular/pull/7628#discussion_r444686832. @simeonoff, as you can see the value, is an object with keys holding the class names. However, in that case, where we have more like an `if-else` case, maybe it is clear to use that expression `[ngClass]="{ items.length > 0 ?  'igx-drop-down__list': 'igx-drop-down__list--empty'}"` ?

@simeonoff, as you've also said the other option is to have a getter with `@HostBinding('attr.class')`. It seems both options can be applied, and I prefer the first one only because the expression that is used is short and don't need to be in the .ts file.